src/components/Main.jsx

```diff
@@ -68,6 +68,22 @@ const Main = () => {
     return <ClassRoster componentNavigationHandler={switchComponents} />;
   };
 
+  useEffect(() => {
+    const sendHeight = () => {
+      // Send height to the parent window
+      const height = document.body.scrollHeight;
+      const message = { type: 'iframeHeight', height };
+      const origingUrl = window.location.origin;
+      window.parent.postMessage(message, origingUrl);
+    };
+
+    const observer = new MutationObserver(() => {
+      sendHeight();
+    });
+
+    observer.observe(document.body, { childList: true, subtree: true });
```

Comment:
I would recommend to disconnect the observer to avoid to have multiple instances:

```suggestion
    observer.observe(document.body, { childList: true, subtree: true });
    
    return () => {
    observer.disconnect(); 
  };
```